cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -239,28 +239,81 @@ OutNode getAnOutNode(DataFlowCall call, ReturnKind kind) {
     )
 }
 
+private predicate writesToGlobal(StoreInstruction store, GlobalOrNamespaceVariable v) {
+  v = store.getResultAddress().(VariableAddressInstruction).getAstVariable()
+}
+
+private predicate loadsFromGlobal(LoadInstruction load, GlobalOrNamespaceVariable v) {
+  v = load.getSourceAddress().(VariableAddressInstruction).getAstVariable()
+}
+
+/** Holds if `i1` is always executed after `i2` */
+bindingset[i1, i2]
+private predicate instrStrictlyPostDominates(Instruction i1, Instruction i2) {
+  i1.getBlock().strictlyPostDominates(i2.getBlock())
+  or
+  exists(IRBlock block, int index1, int index2 |
+    block = i1.getBlock() and
+    block = i2.getBlock() and
```

Comment:
Are these two lines necessary?

Reply:
You're right. They're implied by the next to lines. Good catch!

Reply:
The same change can be applied to `instrStrictlyPostDominates` as well.